publication-commons/src/main/java/no/unit/nva/publication/service/impl/ResourceService.java

```diff
@@ -268,7 +272,15 @@ public Publication getPublication(Publication sampleResource) throws NotFoundExc
     }
 
     public Resource getResourceByIdentifier(SortableIdentifier identifier) throws NotFoundException {
-        return readResourceService.getResourceByIdentifier(identifier);
+        if (shouldUseNewFiles()) {
+            return getResourceAndFilesByIdentifier(identifier).orElseThrow();
+        } else {
+            return readResourceService.getResourceByIdentifier(identifier);
+        }
+    }
+
+    public boolean shouldUseNewFiles() {
```

Comment:
Even if this is temporary, I think it would be nice to put this as a standalone FeatureFlags class maybe?

Reply:
What do you mean?

Reply:
if it can be separated it probably should? Would have the benefit to also mark why its here (its a feature flag). new FeatureFlags(new Environment()).isEnabled("SHOULD_USE_NEW_FILES") or something